### PR TITLE
Support buttons as actions in Page Header

### LINKED ISSUE #1265: PageHeader primaryAction does not support `disable` prop
**Is your feature request related to a problem? Please describe.**
I need to disable the primary action of a PageHeader until something is ready in my app.

**Describe the solution you'd like**

It would be cool to support all button props in `PageHeader.primaryAction` and `PageHeader.secondaryActions`.

**Describe alternatives you've considered**
-

**Additional context**
-

### PR DESCRIPTION
**Pull request by francocorreasosa**
Closes #1265

### CODE REVIEW

**Comment by francocorreasosa** [comment]
@siddharthkp 	Can you review & merge if everything's alright? thanks 🙇

File: core/components/_helpers/button-prop.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import Button from '../atoms/button'
+export default (prop, overrides = {}) => {
```
Comment (siddharthkp) [inline core/components/_helpers/button-prop.js]:
```diff
- prop
+ props
```

File: core/components/_helpers/button-prop.js [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+  if (!prop) return null
+  if (prop.type === Button) return React.cloneElement(prop, overrides)
+
+  if (typeof prop === 'object') {
+    return (
+      <Button icon={prop.icon} onClick={prop.handler} {...overrides}>
```
Comment (siddharthkp) [inline core/components/_helpers/button-prop.js]: We should proxy through all of the props (not just icon and onClick)

```jsx
<Button icon={props.icon} onClick={props.handler} {...props} {...overrides}>
```

File: core/components/_helpers/button-prop.js (hunk 1 above)
Comment (francocorreasosa) [reply core/components/_helpers/button-prop.js]: It is actually one prop (either `primaryAction` or `secondaryAction` in the PageHeader case)

Comment (siddharthkp) [reply core/components/_helpers/button-prop.js]: then let's rename it to `action`

`prop` doesn't communicate the intent

File: core/components/_helpers/button-prop.js (hunk 2 above)
Comment (francocorreasosa) [reply core/components/_helpers/button-prop.js]: Should we? This way of using the props is being deprecated, maybe using other buttons props can be an incentive to move to the new API.

**Comment by francocorreasosa** [comment]
Merging this as-is, and then we can add the remaining case if we consider is needed.

**Pull request merged**